hand-written.md

```diff
@@ -80,66 +41,51 @@ The GenBCode backend includes a new inliner and bytecode optimizer.
 
 * Inlining final methods, including methods defined in objects and final methods defined in traits
 * If a closure is allocated and invoked within the same method, the closure invocation is replaced by an invocations of the corresponding lambda body method
 * Dead code elimination and a small number of cleanup optimizations
+* Box/unbox elimitation [#4858](https://github.com/scala/scala/pull/4858)
 
 The work on the new optimizer is still ongoing.  You can track it in the [scala-dev repository issue tracker](https://github.com/scala/scala-dev/labels/t%3Aoptimizer).
 
-#### The `@implicitAmbiguous` annotation
 
-The `@implicitAmbiguous` annotation allows customizing the error message when an implicit search finds multiple ambiguous values. Refer to the [Scaladoc of the annotation class](http://www.scala-lang.org/api/2.12.x/#scala.annotation.implicitAmbiguous) for an example.
+### Breaking changes
 
-Thanks to @puffnfresh for this contribution!
+#### SAM types
+As of [#4971](https://github.com/scala/scala/pull/4971), we treat Single Abstract Method types in the same way as our built-in FunctionN classes. This means overloading resolution has more contenders to choose from, making type inference less effective. Please consult the PR for details.
 
-#### Assorted improvements to `Future` and `Promise`
+#### Changed syntax trees (affects macro and compiler plugin authors)
 
-New methods, performance improvements, better doc, deprecations, and
-more.  Details at https://github.com/scala/scala/pull/4042
+PR [#4794](https://github.com/scala/scala/pull/4749) changed the syntax trees for selections of statically accessible symbols. For example, a selection of `Predef` no longer has the shape `q"scala.this.Predef"` but simply `q"scala.Predef"`. Macros and compiler plugins matching on the old tree shape need to be adjusted.
 
-Thanks to @viktorklang for this contribution!  (Based, he says, on
-"years" of collected feedback from the Scala community.)
 
-### Unbundled features
 
-The following modules have been removed from the Scala 2.12 distribution:
 
-* Scala standard library actors.
-  We recommend [Akka actors](http://akka.io/) instead.
-  See the [Scala actors migration guide](http://docs.scala-lang.org/overviews/core/actors-migration-guide.html).
-* Akka actors.
-  The Scala distribution and the `scala-library-all` dependency no longer include Akka actors.
-  To use Akka, [add it as a dependency](http://doc.akka.io/docs/akka/2.4.0/intro/getting-started.html).
-* Continuations plugin.
-  ([Community maintainers sought](https://github.com/scala/scala-continuations).)
+## Binary compatibility
 
-### Breaking changes
+Since Scala 2.11, minor releases of Scala are binary compatible with each other.
+Scala 2.12 will continue this tradition: every 2.12.x release will be binary compatible with 2.12.0.
+Milestone releases and release candidates, however, are **not** binary compatible with any other release.
```

Comment:
maybe s/not/not necessarily/, since M5 and RC1 might actually be compatible. I dunno, maybe it's fine as is.